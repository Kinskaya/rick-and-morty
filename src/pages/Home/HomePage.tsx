import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import SearchBar from '../../components/UI/input/SearchBar';
import CardList from '../../components/CardList/CardList';
import { IState } from '../../components/App/App';
import SelectCards from '../../components/UI/SelectCards/SelectCards';
import Pagination from '../../components/UI/Pagination/Pagination';

import { sortCards, fetchData, paginate } from '../../redux/cardsSlice';
import { AppDispatch, RootState } from '../../redux/store';

import './HomePage.css';

const HomePage = () => {
  const dispatch = useDispatch<AppDispatch>();

  const [sort] = useState('');
  const selectCards = () => dispatch(sortCards({ sort }));

  const { selectValue, cardsPerPage, totalCards } = useSelector<RootState>(
    (state) => state.cards
  ) as IState;

  return (
    <div className="wrapper">
      <h1>Rick and Morty</h1>
      <SearchBar addSearch={fetchData} />
      <SelectCards defaultValue="" value={selectValue} onchange={selectCards} />
      <CardList />
      <Pagination cardsPerPage={cardsPerPage} totalCards={totalCards} paginate={paginate} />
    </div>
  );
};

export default HomePage;
