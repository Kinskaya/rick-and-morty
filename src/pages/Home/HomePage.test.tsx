import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import axios from 'axios';
import HomePage from './HomePage';

jest.mock('axios');

async function addHandler(value: string) {
  const response = await axios.get(`https://rickandmortyapi.com/api/character/?name=${value}`);
  return response.data[0].name;
}

describe('HomePage component', () => {
  it('render correctly', () => {
    const { asFragment } = render(<HomePage />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders page title name', () => {
    const { getByText } = render(<HomePage />);
    const linkElement = getByText(/Rick and Morty/i);
    expect(linkElement).toBeInTheDocument();
  });
  it('returns the name of the first item', async () => {
    (axios.get as jest.Mock).mockResolvedValue({
      data: [
        {
          id: 1,
          name: 'Rick Sanchez',
          status: 'Alive',
        },
        {
          id: 2,
          name: 'Morty Smith',
          status: 'Alive',
        },
      ],
    });

    const name = await addHandler('r');
    expect(name).toEqual('Rick Sanchez');
  });
});
