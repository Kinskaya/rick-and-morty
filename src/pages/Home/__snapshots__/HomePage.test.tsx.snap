// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component render correctly 1`] = `
<DocumentFragment>
  <div
    class="wrapper"
  >
    <h1>
      Rick and Morty
    </h1>
    <form>
      <i
        class="material-icons"
      >
        search
      </i>
      <input
        class="search-input"
        placeholder="Search..."
        type="text"
        value=""
      />
    </form>
    <div
      class="card-container"
    />
    <div
      class="modal"
    >
      <div
        class="modal-content"
      >
        <div
          class="card"
        >
          <img
            alt="item-image"
            class="card-image"
            src=""
          />
          <p
            class="title"
          >
            Name: 
            <span />
          </p>
          <p
            class="title"
          >
            Gender: 
            <span />
          </p>
          <p
            class="title"
          >
            Status: 
            <span />
          </p>
          <p
            class="title"
          >
            Species: 
            <span />
          </p>
        </div>
        <button
          class="modal-btn"
        >
          x
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
